plugins/ercc-vscc/ercc_validation_logic.go

```diff
@@ -187,10 +186,15 @@ func (t *VSCCERCC) checkAttestation(respPayload *peer.ChaincodeAction) error {
 		// get mrenclave from ledger
 		mrenclave, err := state.GetState(eccNamespace, MrEnclaveStateKey)
 		if err != nil {
-			return errors.New("mrenclave does not exist")
+			return fmt.Errorf("getting MRENCLAVE failed, err %s", err)
 		}
+		// if mrenclave not already on the ledger
 		if mrenclave == nil {
-			return errors.New("mrenclave is empty")
+			// it must be in the writeset
+			mrenclave = getMrEnclaveWriteSet(eccNamespace, txRWSet)
+			if mrenclave == nil {
+				return fmt.Errorf("no MRENCLAVE on the ledger nor in write set for %s", eccNamespace)
+			}
 		}
```

Comment:
Why is this actually not the only/main case?  Do we currently allow updates? Otherwise, there shouldn't be anything in the state anyway? If so, how are we doing updates securely?  I realize this plugin will go away but the validation logic should transfered to some extent to ercc itself, so this code fragment is still somewhat relevant? maybe some comments could be helpful for the future?

Reply:
Well, the code also considers the case where setup is already called for another enclave , in that case mrenclave may already exists on the ledger. However, I guess this peace of logic needs bigger rework as we will consume MRENCLAVE from chaincode definition and not from ledger state. Thus, I don't think we need to touch it here or add any comment.

Reply:
I added a pointer to this comment in #134